modules/library/jdbc/src/main/java/org/geotools/jdbc/JDBCDataStore.java

```diff
@@ -1557,6 +1559,18 @@ protected Object getAggregateValue(FeatureVisitor visitor, SimpleFeatureType fea
         }
     }
 
+    /**
+     * Checks if the groupby is a supported one, that is, if it's possible to turn to SQL the
+     * various {@link Expression} it's using
+     * @param visitor
+     * @return
+     */
+    private boolean isSupportedGroupBy(GroupByVisitor visitor) {
+        // right now everything must be an attribute
+        return visitor.getExpressions().stream().allMatch(xp -> xp instanceof AttributeDescriptor) 
+            && visitor.getGroupByAttributes().stream().allMatch(xp -> xp instanceof AttributeDescriptor); 
```

Comment:
Not sure if I understand this one, so this basically means it will not be possible to use the data difference function ?

Reply:
This code just updates the checks to the current status, to avoid the exceptions. A new pull request will follow to allow encoding the function in postgresql, that requires a larger number of changes, filter capabilities, a way to check if an expression is supported by the capabilities (right now can only check filter), postgresql specific encoding and so on.

Reply:
Ok makes sense, thank you.